### PR TITLE
Windows install and play batch scripts

### PR DESCRIPTION
**Pull request by Skhmt**
The [install.bat](https://github.com/cloveranon/Clover-Edition/blob/windows-install-run-batch/install.bat) will:

- Download python and put it into /venv
- Download pip and install it into /venv
- Set the proper paths for python39._pth
- Ask you to install the CUDA or CPU version of PyTorch and install that version
- Install Transformers 2.3.0
- Install Prompt_toolkit
- Search for Windows Terminal, and if it isn't found, will download and start the installer for you

The [play.bat](https://github.com/cloveranon/Clover-Edition/blob/windows-install-run-batch/play.bat) will:
- Boot up windows terminal if it's found, or powershell if not
- Run the game with the proper python.exe from /venv/

Did a lot of testing on these, but could use a little more. The tl;dr for Windows 10 x64 users from nothing installed to in-game is:
Download the repo, run install.bat, get a model, then run play.bat

Caveat - this assumes at least Windows 10 version 1903 from January 2018 (that was 3 years ago), because it uses `curl` and `tar`.
If they're on an older version of windows... I don't know.

### CODE REVIEW

**Comment by Meanmelter** [comment]
I can confirm this script 100% works for my machine

Curl/tar has been a feature since 1803, Anyone with Windows older / Windows 7 will not be able to run
We could switch to PowerShell to install python via Net.WebClient As far as I know this is a feature even on Windows 7 PowerShell

**Comment by Skhmt** [comment]
Yeah, I was thinking about writing some extra logic to check if curl/tar are there, and switch to powershell cmdlets if they aren't there.

**Comment by Meanmelter** [comment]
Also, I do like the idea of having our own python environment, lets you keep your own (locally installed) version of python, and keeps Clover to using it's own which we can easily update.  Makes versioning easy as well since we can just reference local and update what is needed

It's a bit hard in .BAT files, but if you do end up calling PowerShell it's pretty easy to check download files by doing a hash check just to make sure it's not corrupted or didn't download fully.

**Comment by Skhmt** [comment]
Added fallbacks to powershell Invoke-WebRequest instead of curl and Expand-Archive instead of tar. If they don't have either of those, they might not even be on Windows 7.

**Comment by Meanmelter** [comment]
> You can check a filehash in a .bat with certutil -hashfile myfile.zip MD5 or SHA1 or SHA256 or SHA512.
Yes, you may have more luck with it than me! Lol, I had a hard time getting it to work two years ago running a job for a client. Ended up just using PowerShell
> There's another reason too, which is insane. Installing Python via the Windows Store adds `pip` to your PATH, and I _think_ `python`, but not `py`. Installing Python via the official installer from python.org does _NOT_ add `pip` nor `python` to your PATH by default - it instead adds `py`, which does what `python` does, but is named differently. So depending on where you install it, you might have to run `pip`, `py -m pip`, or `python -m pip`. Also, you can't just check if `python.exe` is in your PATH, because it's always in the PATH for Windows 10 - invoking it brings up the Microsoft Store to install Python. It's a massive headache, so manually downloading the zip and extracting it was actually the easier path.
I usually always install python custom so I can run `pip` from path, very very handy.

**Comment by Skhmt** [comment]
The last feature I was thinking of was incorporating getting a model into the installer. But that would definitely require downloading something like Aria2 since they're all in torrent files. I could just open the mega/gdrive/torrent in their default browser/torrent app instead, but then I can't handle the extraction/moving/naming. Thoughts?

**Comment by SammCheese** [comment]
Might not be Possible but you could try to get the Systems VRAM and show it to the User alongside different Models and their VRAM requirement.

**Comment by Meanmelter** [comment]
I would just use Powershell to download the models if possible. Though Powershell can't handle magnet links without a separate component. Google drive needs an OAuth code that the file host has to generate for it to work with Powershell. 
`Invoke-WebRequest -Uri $Url -OutFile $OutputPath`
Given most are shared via magnet links, I think aria2c is the cleanest option possible.

**Comment by MikkoMMM** [comment]
This needs to be updated for a new transformers version now. Would it be sensible for it to fetch that from the requirements.txt even?

**Comment by Skhmt** [comment]
Oh finetuneanon's PR was accepted? I'll update this tonight or tomorrow.

File: install.bat [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+:: Create /venv/
+echo Creating ./venv/
+if not exist "./venv" mkdir venv
+
+cd venv
+
+:: Download Python
+echo Downloading Python...
+if defined HasCurl (
+  curl "%PythonURL%" -o %cd%\python.zip
```
Comment (AtaIanta) [inline install.bat:51]:
```suggestion
  curl "%PythonURL%" -o "%cd%\python.zip"
```
image
fixes error when folder name has space in it

File: install.bat [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+) else (
+  powershell Invoke-WebRequest -Uri "%PythonURL%" -OutFile %cd%\python.zip
```
Comment (AtaIanta) [inline install.bat:53]:
```suggestion
  powershell Invoke-WebRequest -Uri "%PythonURL%" -OutFile "%cd%\python.zip"
```
ditto

**Pull request merged**